### PR TITLE
Clean up mobile/library/common/jni/BUILD targets

### LINKED ISSUE #24544: Envoy Mobile jni BUILD targets are confusing
`jni_version.cc` and `jni_utility.cc` are listed in srcs for multiple build targets. This ends up with the potential for duplicate symbols. Here's a dependency graph of some of the targets involved.

Screenshot 2022-12-14 at 5 13 02 PM

`jni_interface.cc` is also listed in srcs for multiple build targets. While this doesn't appear to cause problems it's still a bitunorthodox.

Finally, there is an annoying warning that we should clean up:
```
WARNING: /usr/local/google/home/rch/src/envoy/mobile/library/common/jni/BUILD:95:27: target 'libndk_envoy_jni.jnilib' is both a rule and a file; please choose another name for the rule
```

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
Signed-off-by: Ryan Hamilton <rch@google.com>

- Remove unnecessary copts now that c++17 is the default.
- Rename libndk_envoy_jni.jnilib -> libndk_envoy_jni_jnilib to squelch warning
- Use ":base_java_jni_lib" instead of "base_java_jni_lib", per standards
- Depend on jni_utility_lib in base_java_jni_lib instead of duplicating the srcs

Fixes #24544

### CODE REVIEW

File: mobile/library/common/jni/BUILD [hunk 1]
```diff
@@ -95,7 +93,7 @@ android_debug_info(
 ## Targets for local execution
 # OS X binary (.jnilib) for NDK testing
 envoy_mobile_so_to_jni_lib(
-    name = "libndk_envoy_jni.jnilib",
```
Comment (keith) [inline mobile/library/common/jni/BUILD:98]: pretty sure this name is important for testing on macOS, @jpsim or @Augustyniak can verify

Comment (jpsim) [reply mobile/library/common/jni/BUILD:98]: Seeing as how CI runs tests that go through the JNI on macOS, I think that's sufficient coverage, but I'll test locally to be sure.

**Comment by jpsim** [comment]
> Remove unnecessary copts now that c++17 is the default.

Does that apply to the NDK toolchain too? With your branch, if I pass `-s` to a bazel test command, I see both `-std=c++11` and `-std=c++17` in the build arguments, which I find puzzling.

**Comment by keith** [comment]
That's probably bazel's default + our .bazelrc, so as long as the 17 is later it should be fine

**Comment by jpsim** [comment]
> That's probably bazel's default + our .bazelrc, so as long as the 17 is later it should be fine

Yes, the 17 is later.

File: mobile/library/common/jni/BUILD (hunk 1 above)
Comment (jpsim) [reply mobile/library/common/jni/BUILD:98]: `./bazelw test --define=signal_trace=disabled //test/java/...` still passes for me locally on macOS.

**Comment by RyanTheOptimist** [comment]
Thanks for the double-check, @jpsim and thanks for the comments @keith!

**Comment by jpsim** [comment]
I don't see any issues with this from my testing.

While you're here, maybe we can remove this, which appears to have only been needed for Rust, which Lyft doesn't use in Envoy Mobile anymore: https://github.com/envoyproxy/envoy/blob/eeab187b21e540dbe47b83d023b94f4682fb1179/mobile/library/common/jni/BUILD#L80

**Pull request merged**